lib/miq_automation_engine/engine/miq_ae_method.rb

```diff
@@ -86,31 +86,13 @@ def self.invoke_external(cmd, workspace, serialize_workspace = false)
 
       ENV['MIQ_TOKEN'] = ws.guid unless ws.nil?
 
-      rc = nil
-      final_stderr = nil
-      begin
-        require 'open4'
-        status = Open4.popen4(*cmd) do |pid, stdin, stdout, stderr|
-          stdin.close
-          final_stderr = stderr.each_line.map(&:strip)
-          stdout.each_line { |msg| $miq_ae_logger.info  "Method STDOUT: #{msg.strip}" }
-          final_stderr.each { |msg| $miq_ae_logger.error "Method STDERR: #{msg}" }
-        end
-
-        unless ws.nil?
-          ws.reload unless ws.nil?
-          ws.setters.each { |uri, value| workspace.varset(uri, value) } unless ws.setters.nil?
-          ws.delete
-        end
-        rc  = status.exitstatus
-        msg = "Method exited with rc=#{verbose_rc(rc)}"
-      rescue => err
-        $miq_ae_logger.error("Method exec failed because (#{err.class}:#{err.message})")
-        rc = MIQ_ABORT
-        msg = "Method execution failed"
+      rc, msg, final_stderr = run_method(*cmd)
+      if ws
```

Comment:
This now happens even if `popen4` raises... and is itself no longer protected by the `rescue`.

Neither of those sound particularly troubling, but I thought I'd mention it.